Make generated Kotlin protos extend Message

Egorand opened the pull request:
Fixes #706

JakeWharton commented on wire-runtime/src/test/proto-kotlin/com/squareup/wire/protos/kotlin/person/Person.kt:
```diff
@@ -21,7 +24,18 @@ data class Person(
-) {
+) : Message<Person, Person.Builder>(ADAPTER, unknownFields) {
+    @Deprecated(
+            message = "Not relevant for data classes, use copy()",
```
I don't find this to be true since there's no way to have multiple functions contribute to the building of one object.

JakeWharton replied:
Also this is going to cause deprecations on the Java side for all these things where Builder is the only thing you have

Egorand replied:
The idea is to generate different code for Kotlin-only/mixed codebases: for Kotlin-only, we don't really need Builders, but we still have to extend `Message` and provide a `Message.Builder` implementation, deprecation annotations should prevent people from using the Builder and prefer `copy()`; for mixed codebases we'll generate full versions of Builders to remain source-compatible with Java protos - this will be hidden behind a compiler flag.

JakeWharton replied:
In that case they should be marked as HIDDEN without a ReplaceWith

oldergod replied:
Is this `HIDDEN` the solution to https://youtrack.jetbrains.com/issue/KT-24769 ?

mc-lovin replied:
Can kotlin functions with HIDDEN annotations be accessed from JAVA?

mc-lovin commented:
can we add a flag for this ?

Egorand commented:
@mc-lovin Currently protos have to extend `Message`, otherwise Retrofit's `WireConverterFactory` won't work.

Egorand commented:
@JakeWharton I removed `Deprecated` from `Builder` (there's no way to make it `HIDDEN`, cause then the proto itself doesn't compile) and added `HIDDEN` to `newBuilder()`. Not sure this is what makes the most sense, let's change this later if there are better ideas.
